### ISSUE TITLE
Allow Configurable FTPEnvironment.setListHiddenFiles()?

### ISSUE THREAD

**Issue by alessandroscarlatti** [robtimus/ftp-fs#10]
First of all, I am very impressed with your implementation! Thank you for sharing your work!

Curious if you have any thoughts on this...

I was experimenting with ftp-fs with a python FTP server [pyftpdlib](https://github.com/giampaolo/pyftpdlib).

`Files.list(fs.getPath("dir1"))` threw an exception, although the directory did exist.  Upon debugging I learned that for some reason (I have not determined exactly) pyftpdlib does not support FTP command `LIST` with option `-a`. Apparently this is controlled by `FTPClient#setListHiddenFiles(true/false)`.

I did find a workaround:
I manually subclassed FTPEnvironment to instead call `client.setListHiddenFiles(true)`. The `Files.list(...)` invocation worked.

I don't see an explicit comment in the `FTPEnvironment#initializePreConnect()` method on  indicating why we need to `setListHiddenFiles(true)` at this point, but I presume it is intended to set that property in case a user wishes to access hidden files. Have you considered if this setting should be configurable somehow (other than subclassing) since it is not part of the `FTPClientConfig` object?

https://github.com/robtimus/ftp-fs/blob/master/src/main/java/com/github/robtimus/filesystems/ftp/FTPEnvironment.java#L762

```
 void initializePreConnect(FTPClient client) throws IOException {
        client.setListHiddenFiles(true);  // not all FTP clients appear to support this

        if (containsKey(SEND_BUFFER_SIZE)) {
            int size = FileSystemProviderSupport.getIntValue(this, SEND_BUFFER_SIZE);
            client.setSendBufferSize(size);
        }
        ....
```

**Comment by robtimus** [comment]
The list hidden files flag initially needed to be set to true, because directory entry `.` was used. That's still used by `FTPFileStrategyFactory.UNIX`. I set the flag to false and ran the unit tests, and four of them failed, because they expected that `FTPFileStrategyFactory.UNIX` was used but instead (because of auto-detection) it was now `FTPFileStrategyFactory.NON_UNIX` - because directory entry `.` is only returned if hidden files are returned. This is something to take into account.

**Comment by robtimus** [comment]
I've just released version [2.0](https://github.com/robtimus/ftp-fs/releases/tag/2.0) that includes new method `withListHiddenFiles`. Keep in mind that you cannot use the `UNIX` strategy, as that will trigger incorrect `NotDirectoryExceptions`.
